tools/build/docker/Dockerfile

```diff
@@ -29,25 +38,48 @@ RUN \
     adduser -D -u ${LRR_UID} -G koyomi koyomi; \
 fi
 
+
+# we use s6-overlay-nobin to just pull in the s6-overlay arch agnostic (shell)
+# components, since we apk install the binaries of s6 later which are arch specific
+ADD https://github.com/just-containers/s6-overlay/releases/download/${S6_OVERLAY_RELEASE}/s6-overlay-nobin.tar.gz /tmp/s6-overlay-nobin.tar.gz
+RUN tar -C / -xzf /tmp/s6-overlay-nobin.tar.gz && rm -f /tmp/s6-overlay-nobin.tar.gz
+
+
 WORKDIR /home/koyomi/lanraragi
 
 #Copy cpanfile and install script before copying the entire context
 #This allows for Docker cache to preserve cpan dependencies
-COPY --chown=koyomi:koyomi /tools tools
+COPY --chown=koyomi:koyomi /tools/cpanfile /tools/install.pl tools/
 COPY --chown=koyomi:koyomi /package.json package.json
 
-# Make scripts executable + Run the install script as root
-RUN chmod +x ./tools/build/docker/install-everything.sh ./tools/build/docker/entrypoint.sh && sh ./tools/build/docker/install-everything.sh
+# Install deps (we install Linux::Inotify2 manually since it's not in the base cpanfile as it doesn't build on macOS)
+RUN apk add --update perl perl-io-socket-ssl perl-dev redis libarchive-dev libbz2 openssl-dev zlib-dev \
```

Comment:
originally I did it to prevent the tools folder updating and clearing the layer cache, but that doesn't happen anymore. personally I think it's still a good idea because it prevents jumping around different files trying to find what the docker build is actually doing, but I can revert if you want

Reply:
it may also make it easier to seperate out build/test/runtime deps later if we decide to do multistage builds

Reply:
Well with the s6 config files we already need to jump around a bunch of files to figure out how it works now 😅  
Moving things back to a RUN command if we go multistage later shouldn't be too hard, so I'd like a revert for now. 👍

Reply:
that's true, lol. gotcha, reverting